js/XAPI.js

```diff
@@ -1404,12 +1397,14 @@ class XAPI extends Backbone.Model {
   async processAttachments(statement) {
     const attachments = statement.attachments;
 
-    await Async.each(attachments, (attachment, nextAttachment) => {
-
-      // First check the attachment for a value
+    for (let attachment of attachments) {
+      await new Promise((resolve, reject) => {
+        // First check the attachment for a value
       if (attachment.value) {
-        nextAttachment();
-      } else if (attachment.url) {
+        resolve();
```

Comment:
you'll have to return here otherwise it'll carry on into the attachment.url block